crashtracker/src/collector/api.rs

```diff
@@ -161,3 +161,432 @@ fn test_crash() -> anyhow::Result<()> {
     assert_eq!(q, 3);
     Ok(())
 }
+
+#[test]
+fn test_altstack_paradox() -> anyhow::Result<()> {
+    use crate::StacktraceCollection;
+    use chrono::Utc;
+    use ddcommon::Endpoint;
+
+    let time = Utc::now().to_rfc3339();
+    let dir = "/tmp/crashreports/";
+    let output_url = format!("file://{dir}{time}.txt");
+
+    let endpoint = Some(Endpoint::from_slice(&output_url));
+
+    let create_alt_stack = true;
+    let use_alt_stack = false;
+    let resolve_frames = StacktraceCollection::EnabledWithInprocessSymbols;
+    let timeout_ms = 10_000;
+
+    // This should return an error, because we're creating an altstack without using it
+    let config = CrashtrackerConfiguration::new(
+        vec![],
+        create_alt_stack,
+        use_alt_stack,
+        endpoint,
+        resolve_frames,
+        timeout_ms,
+    );
+
+    assert!(
+        config.is_err(),
```

Comment:
Should we check what the particular error is?

Reply:
Yes, good call.  Otherwise we can't guarantee we're testing the precise effect we desired.